plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -395,31 +510,23 @@ public IpDeployer getIpDeployer(Network network) {
 
     @Override
     public boolean applyFWRules(Network network, List<? extends FirewallRule> rules) throws ResourceUnavailableException {
-        s_logger.debug("Handling applyFWRules for network " + network.getName() + " with " + rules.size() + " FWRules");
-        if (rules != null && rules.size() == 1 && rules.iterator().next().getType().equals(FirewallRuleType.System)) {
+        if (rules.size() == 1 && rules.iterator().next().getType().equals(FirewallRuleType.System)) {
             s_logger.debug("Default ACL added by CS as system is ignored for network " + network.getName() + " with rule " + rules);
             return true;
         }
-        return applyACLRules(network, rules, false);
-    }
 
-    @Override
-    public boolean applyNetworkACLs(Network network, List<? extends NetworkACLItem> rules) throws ResourceUnavailableException {
-        if (rules == null || rules.isEmpty()) {
-            s_logger.debug("No rules to apply. So, delete all the existing ACL in VSP from Subnet with uuid " + network.getUuid());
-        } else {
-            s_logger.debug("New rules has to applied. So, delete all the existing ACL in VSP from Subnet with uuid " + network.getUuid());
-        }
-        if (rules != null) {
-            s_logger.debug("Handling applyNetworkACLs for network " + network.getName() + " with " + rules.size() + " Network ACLs");
-            applyACLRules(network, rules, true);
+        if (rules.isEmpty()) {
+            return true;
         }
-        return true;
+
+        s_logger.debug("Handling applyFWRules for network " + network.getName() + " with " + rules.size() + " FWRules");
```

Comment:
Wrap in a if (s_logger.isDebugEnabled()) { ... } block to avoid unnecessary pressure on the string pool when debug logging is not enabled.